dms-service/src/main/java/de/muenchen/oss/swim/dms/domain/helper/MetadataHelper.java

```diff
@@ -0,0 +1,83 @@
+package de.muenchen.oss.swim.dms.domain.helper;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import de.muenchen.oss.swim.dms.configuration.SwimDmsProperties;
+import de.muenchen.oss.swim.dms.domain.exception.MetadataException;
+import de.muenchen.oss.swim.dms.domain.model.DmsTarget;
+import jakarta.validation.constraints.NotNull;
+import java.io.IOException;
+import java.io.InputStream;
+import lombok.RequiredArgsConstructor;
+import org.apache.logging.log4j.util.Strings;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class MetadataHelper {
+    public static final String METADATA_VALUE_KEY = "Value";
+    public static final String METADATA_KEY_KEY = "Name";
+    public static final String METADATA_INDEX_FIELDS_KEY = "IndexFields";
+
+    private final SwimDmsProperties swimDmsProperties;
+    private final ObjectMapper objectMapper;
+
+    /**
+     * Extract dms target from metadat file.
+     *
+     * @param inputStream InputStream of metadata file.
+     * @return The dms target.
+     * @throws MetadataException If file can't be parsed or required values are missing.
+     */
+    public DmsTarget resolveDmsTarget(@NotNull final InputStream inputStream) throws MetadataException {
+        try {
+            final JsonNode rootNode = objectMapper.readTree(inputStream);
+            final JsonNode indexFieldsNode = rootNode.get(METADATA_INDEX_FIELDS_KEY);
+            String userInboxCoo = null;
+            String userInboxOwner = null;
+            String groupInboxCoo = null;
+            String groupInboxOwner = null;
+            for (final JsonNode indexField : indexFieldsNode) {
+                // user inbox coo
+                if (swimDmsProperties.getMetadataUserInboxCooKey().equals(indexField.path(METADATA_KEY_KEY).asText())) {
```

Comment:
Kann "asText" nicht potentiell auf einen NullPointer laufen, wenn das Metadata-File keinen solchen Key enthält? Ich sehe aus der JavaDoc von path, dass dann ein  "missing node" kommt, aber was macht "asText" dann daraus?

Reply:
Die Beschreibung vom .path sagt folgendes: `This method is similar to get(String), except that instead of returning null if no such value exists (due to this node not being an object, or object not having value for the specified field), a "missing node" (node that returns true for isMissingNode) will be returned. This allows for convenient and safe chained access via path calls.` sollte meiner Meinung nach also passen.
Hab auch einen Test eingebaut. Falls "IndexFields" fehlt, dann kommt es zu einem NullPointer, aber denke das passt.